tests/test_typescript_config.py

```
@@ -0,0 +1,353 @@
+    """Repeated imports from one source tree reuse discovered config paths."""
+    importer = _write(tmp_path / "src" / "main.ts")
+    first = _write(tmp_path / "src" / "first.ts")
+    second = _write(tmp_path / "src" / "second.ts")
+    _write(
+        tmp_path / "tsconfig.json",
+        '{"compilerOptions":{"baseUrl":".","paths":{"@/*":["src/*"]}}}',
+    )
+    original_walk = os.walk
+    walk_calls = 0
+
+    def counting_walk(*args, **kwargs):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `counting_walk`

(ANN202)